src/ui/component/fileList/view.jsx

```diff
@@ -1,165 +1,66 @@
 // @flow
 import * as React from 'react';
-import { buildURI, SORT_OPTIONS } from 'lbry-redux';
-import { FormField, Form } from 'component/common/form';
-import FileCard from 'component/fileCard';
+import FileListItem from 'component/fileListItem';
+import Spinner from 'component/spinner';
+import { FormField } from 'component/common/form';
+import usePersistedState from 'util/use-persisted-state';
+
+const SORT_NEW = 'new';
+const SORT_OLD = 'old';
 
 type Props = {
-  hideFilter: boolean,
-  sortByHeight?: boolean,
-  claimsById: Array<StreamClaim>,
-  fileInfos: Array<FileListItem>,
-  sortBy: string,
-  page?: string,
-  setFileListSort: (?string, string) => void,
+  uris: Array<string>,
+  header: React.Node,
+  sort: React.Node,
+  injectedItem?: React.Node,
+  loading: boolean,
+  // If using the default header, this is a unique ID needed to persist the state of the filter setting
+  defaultHeader?: string,
```

Comment:
I agree it was pretty messy. I added `persistedStorageKey` and use the default header if no `header` is passed in. If no `peristedStorageKey` is passed in, it will share state with other instances of the component that don't pass it in.

I added a key for every component right now.